pyramid_swagger/tween.py

```diff
@@ -302,24 +334,42 @@ def build_swagger_handler(settings, schema):
     handle both Swagger 1.2 and Swagger 2.0.
 
     :type settings: dict
-    :type schema: :class:'
+    :type schema: :class:`pyramid_swagger.model.SwaggerSchema`
     :rtype: :class:`SwaggerHandler`
     """
     swagger_versions = get_swagger_versions(settings)
 
     if SWAGGER_20 in swagger_versions:
-        return SwaggerHandler(
-            op_for_request=get_op_for_request,
-            handle_request=swaggerize_request,
-            handle_response=swaggerize_response,
-        )
+        return build_swagger20_handler()
 
     if SWAGGER_12 in swagger_versions:
-        return SwaggerHandler(
-            op_for_request=schema.validators_for_request,
-            handle_request=handle_request,
-            handle_response=validate_response,
-        )
+        return build_swagger12_handler(schema)
+
+
+def build_swagger20_handler():
+    return SwaggerHandler(
+        op_for_request=get_op_for_request,
+        handle_request=swaggerize_request,
+        handle_response=swaggerize_response,
+    )
+
+
+def build_swagger12_handler(schema):
+    return SwaggerHandler(
+        op_for_request=schema.validators_for_request,
+        handle_request=handle_request,
+        handle_response=validate_response,
+    )
+
+
+def build_swagger12_handler_wrapper(schema):
```

Comment:
This wrapper is needed to check that schema is non-null. It could be `None` if swagger_versions only had `2.0` and `build_swagger12_handler_wrapper` is called. It should simply return `None` in this case.